internal/deploy/deploy.go

```diff
@@ -0,0 +1,105 @@
+package deploy
```

Comment:
Would be better to rename file `deploy.go` to `deploy-bundle.go` in the deploy dir.

Reply:
I made it generic in purpose so we can later on move some of other deployment such of NFD to use that internal package instead of duplicating code. 

It can also be mocked because it uses an interface (unlike other packages) which will highly be easier to unit-tests other packages using this one.

At first I was trying to do also converge NFD to start using this package (by adding the subscription deployment methods) but the PR was huge so I started with the minimal needs for bundle.